chapters/data-processing.tex

```diff
@@ -0,0 +1,784 @@
+Be mindful of how and where these files are stored
+as they cannot be re-created and nearly always contain confidential information.
+Raw data files must \textit{never} be edited directly.
+If the raw data contains confidential information,
+any tasks that precede de-identification must be carried on encrypted data.
+ 
+% Why cleaning is necessary
+Raw data files may be received in a variety of formats, 
+and are often not easy to handle using statistical software.
+The data cleaning tasks described in this chapter will prepare the raw data
+for quantitative analysis.
+To make this complex process less error-prone,
+we recommend that no changes be made to the content of data at this point
+(with one exception that will be carefully discussed).
+Correcting outliers, imputing values and creating derived indicators
+are all research decisions, and are therefore considered part of analysis.
+Therefore, the main output of data cleaning, the cleaned dataset,
+contains almost exactly same information as the raw data,
+but in a format that is easier to interact with.  
+
+\section{Identifying the identifier(s)}
+
+% Intro
+The very first step in creating statistical software-friendly data sets
+is understanding the data acquired,
+and translating this understanding into an intuitive format.
+This section discusses what steps may be needed to make sure that each row
+in a dataset represents one observation.
+This may seem trivial, but should not be taken for granted.
+Getting to such a format may be harder than expected,
+and the \textbf{unit of observation}\sidenote{\textbf{Unit of observation:}
+	the unit described by the data. \url{
+		https://dimewiki.worldbank.org/wiki/Unit_of_Observation}}
+ may be ambiguous in many raw datasets.
+But having data in a tidy format is the best way to manipulate it, 
+particularly aggregating and cross-referencing.
+You will need to do manipulate data as soon as it is received,
+so you can assure its quality.
+If you are conducting primary data collection,
+you can start coding the data tidying before the data is acquired,
+since you will know in advance in which format it will be received.
+
+\subsection{Unique identifiers}
+
+% Uniquely and fully identifying variable
+The first step in tidying data is to understand its unit of observation (what makes a row),
+and find out which variable or set of variables is the \textbf{unique identifier}\sidenote{\url{
+		https://dimewiki.worldbank.org/ID\_Variable\_Properties}} for each row.
+Ensuring that observations are uniquely and fully identified
+is possibly the most important step in data cleaning.
+It may be the case that the variables expected to be unique identifiers are either incomplete or contain duplicates.
+It is also possible for a data set to have no unique identifier, or for the identifier involves a long string, such as a name.
+In such cases, cleaning begins by carefully creating a variable that uniquely identifies the data.
```

Comment:
specify that the id variable should be numeric?

Reply:
I think we discussed in relation to book or something else that string IDs are ok. Like `AB1234` or `12B42A` etc.